### PR TITLE
IGNITE-19401 Sql. Correct processing count(distinct) without arguments

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

**Comment by korlov42** [comment]
To be honest, I didn't get this patch. Could you please sed some light on the root cause of the problem and the idea of proposed solution?

File: modules/sql-engine/src/integrationTest/sql/subquery/scalar/test_grouped_correlated_subquery.test [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # name: test/sql/subquery/scalar/test_grouped_correlated_subquery.test
 # description: Test correlated subquery with grouping columns
 # group: [scalar]
+# Ignore https://issues.apache.org/jira/browse/IGNITE-22260
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/sql/subquery/scalar/test_grouped_correlated_subquery.test:4]: we already mention these tickets for particular tests, no any reasons to repeat it here

File: modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # name: test/sql/subquery/any_all/test_correlated_any_all.test
 # description: Test correlated ANY/ALL subqueries
 # group: [any_all]
-# Ignore https://issues.apache.org/jira/browse/IGNITE-19401
+# Ignore https://issues.apache.org/jira/browse/IGNITE-22260
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test:4]: We agreed to close the ticket. our current behavior we consider as correct

Comment (zstan) [reply modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test]: got it

File: modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test [hunk 3]
```diff
@@ -157,15 +146,13 @@ true
...
-# Ignore https://issues.apache.org/jira/browse/IGNITE-19401
-skipif ignite3
 query T
 SELECT i=ALL(SELECT i FROM integers WHERE i=i1.i OR i IS NULL) FROM integers i1 ORDER BY i;
 ----
-NULL
-NULL
-NULL
-NULL
+true
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test:152]: I don't understand why you change expected results here

File: modules/sql-engine/src/integrationTest/sql/subquery/scalar/test_grouped_correlated_subquery.test (hunk 1 above)
Comment (ygerzhedovich) [reply modules/sql-engine/src/integrationTest/sql/subquery/scalar/test_grouped_correlated_subquery.test]: you fixed not for all cases

File: modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test (hunk 3 above)
Comment (ygerzhedovich) [reply modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test:152]: MariaDb and PG returns 4 nulls in this case, as was in initial test

Comment (ygerzhedovich) [reply modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test:152]: yes, I have the similar understanding and it looks as bug

**Comment by zstan** [comment]
@ygerzhedovich seems i fix all your issues, plz check once more?

File: modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test (hunk 3 above)
Comment (zstan) [reply modules/sql-engine/src/integrationTest/sql/subquery/any_all/test_correlated_any_all.test]: fill the issue, annotate

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java [hunk 4]
```diff
@@ -272,7 +274,9 @@ private final class AccumulatorWrapperImpl implements AccumulatorWrapper<RowT> {
             this.accumulator = accumulator;
             this.inAdapter = inAdapter;
             this.outAdapter = outAdapter;
+            distinct = call.isDistinct();
 
+            // need to be refactored after https://issues.apache.org/jira/browse/CALCITE-5969
```
[the remark this answers is not in the document]
Comment (zstan) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java]: got it, done

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**